src/plugins/intel_cpu/src/nodes/rdft.h

```diff
@@ -94,6 +95,65 @@ struct RDFTExecutor {
     std::vector<float> generateTwiddlesCommon(size_t inputSize, size_t outputSize, enum dft_type type, bool useFFT);
 };
 
+struct RDFTRefExecutor : public RDFTExecutor {
+    RDFTRefExecutor(bool inverse);
+
+private:
+    std::vector<float> generateTwiddlesDFT(size_t inputSize, size_t outputSize, enum dft_type type) override;
+    void dftRealToComplex(float* inputPtr,
+                          const float* twiddlesPtr,
+                          float* outputPtr,
+                          size_t inputSize,
+                          size_t outputSize,
+                          bool parallelize);
+
+    void dftComplexToComplex(float* inputPtr,
+                             const float* twiddlesPtr,
+                             float* outputPtr,
+                             size_t inputSize,
+                             size_t signalSize,
+                             size_t outputSize,
+                             bool parallelize);
+
+    void dftComplexToReal(float* inputPtr,
+                          const float* twiddlesPtr,
+                          float* outputPtr,
+                          size_t inputSize,
+                          size_t signalSize,
+                          size_t outputSize,
+                          bool parallelize);
+
+    void dft(float* inputPtr,
+             const float* twiddlesPtr,
+             float* outputPtr,
+             size_t inputSize,
+             size_t signalSize,
+             size_t outputSize,
+             enum dft_type type,
+             bool parallelize) override;
+};
+
+#if defined(OPENVINO_ARCH_X86_64)
+struct RDFTJitExecutor : public RDFTExecutor {
+    RDFTJitExecutor(bool inverse, NodeDesc* primDesc = nullptr);
+
+    std::vector<float> generateTwiddlesDFT(size_t inputSize, size_t outputSize, enum dft_type type) override;
+    void dft(float* inputPtr,
+             const float* twiddlesPtr,
+             float* outputPtr,
+             size_t inputSize,
+             size_t signalSize,
+             size_t outputSize,
+             enum dft_type type,
+             bool parallelize) override;
+
+    std::unique_ptr<jit_dft_kernel> rdftKernel = nullptr;
+    std::unique_ptr<jit_dft_kernel> dftKernel = nullptr;
+
+    int vlen;
```

Comment:
Should be these fields in private section?

Reply:
Can be, moved to private.